delivery_location/tests/test_deliverylocation.py

```diff
@@ -0,0 +1,112 @@
+import pytest
+from delivery_location.models import Supplier, DeliveryLocation
+from django.core.exceptions import ValidationError
+import datetime
+
+
+@pytest.fixture
+def TEST_DATA():
+    return [
+        ('taxas', 'firstname', 'lastname', 'password', 'supplier', 'business name'),
+        ('ed', 'edd', 'eddy', 'jawbreaker101', 'supplier', 'jawbreaker inc'),
+    ]
+
+
+@pytest.fixture
+def supplier0(TEST_DATA):
+    return Supplier(TEST_DATA[0])
+
+
+@pytest.fixture
+def delivery_location0(supplier0):
+    return DeliveryLocation(user_name=supplier0, location="Qiryat Shemona", date=datetime.date(2022, 12, 30))
+
+
+@pytest.fixture
+def delivery_location1(supplier0):
+    return DeliveryLocation(user_name=supplier0, location="Haifa", date=datetime.date(2022, 12, 31))
+
+
+class TestDeliveryLocationModel:
+    @pytest.mark.django_db()
+    def test_add_delivery_location(self, supplier0, delivery_location0, delivery_location1):
+        supplier0.save()
+        delivery_location0.add_delivery_location()
+        delivery_location1.add_delivery_location()
+        assert delivery_location0 in DeliveryLocation.objects.all()
+        assert delivery_location1 in DeliveryLocation.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_non_existing_delivery_location(self):
+        with pytest.raises(DeliveryLocation.DoesNotExist):
+            DeliveryLocation.remove_delivery_location_by_id(-1)
+
+    @pytest.mark.django_db()
+    def test_delete_delivery_location(self, supplier0, delivery_location0):
+        supplier0.save()
+        delivery_location0.add_delivery_location()
+        assert delivery_location0 in DeliveryLocation.objects.all()
+        delivery_location0.remove_delivery_location()
+        assert delivery_location0 not in DeliveryLocation.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_id(self, supplier0, delivery_location0):
+        supplier0.save()
+        delivery_location0.add_delivery_location()
+        assert delivery_location0 == DeliveryLocation.get_delivery_location_by_id(delivery_location0.id)
+        with pytest.raises(DeliveryLocation.DoesNotExist):
```

Comment:
This should be in a test case of its own

Reply:
fixed